ddmark/validation.go

```diff
@@ -241,3 +297,53 @@ func parseIntOrUInt(value reflect.Value) (int, bool) {
 
 	return fieldInt, ok
 }
+
+// checkValueExistOrIsValid checks if a given string marker item name value exist in a unmarshalled struct (converted to a map by structValueToMap)
+// it returns 1 if the value is found and -if applicable- the required value is valid, 0 otherwise
+func checkValueExistsOrIsValid(markerItem string, structMap map[string]interface{}, ruleName string) (int, error) {
+	// marketItem can either be a fieldName, or fieldName=fieldValue
+	markerItemArr := strings.Split(markerItem, "=")
+
+	switch len(markerItemArr) {
+	// no given value to respect => check item is not null
+	case 1:
+		if structMap[markerItem] != nil {
+			return 1, nil
+		}
+	// a value was required => check item has described value
+	case 2:
+		markerSubfieldName, markerSubfieldValue := markerItemArr[0], markerItemArr[1]
+		if structMap[markerSubfieldName] == nil {
+			break
+		}
+
+		if reflect.Indirect(reflect.ValueOf(structMap[markerSubfieldName])).Type().ConvertibleTo(reflect.TypeOf(markerSubfieldValue)) {
```

Comment:
There is a lot of nesting here that makes the "golden path" hard to understand.

In general you should favour escaping when it's not the golden path to reduce nesting.

More details:
* https://www.youtube.com/watch?v=CFRhGnuXG-4&ab_channel=CodeAesthetic
* https://drew.thecsillags.com/posts/2014-04-14-refactoring-flow-control-and-conditionals/

Reply:
I would even extract this to a method TBH (extract or even compare int or string or something similar)

Reply:
Came from a switch refactoring -- just realised how horrific that code was, adjusting !